src/main/java/org/apache/maven/plugins/dependency/tree/DOTDependencyNodeVisitor.java

```diff
@@ -47,29 +48,38 @@ public DOTDependencyNodeVisitor(Writer writer) {
      */
     @Override
     public boolean visit(DependencyNode node) {
-        if (node.getParent() == null || node.getParent() == node) {
-            writer.write("digraph \"" + node.toNodeString() + "\" { " + System.lineSeparator());
-        }
-
-        // Generate "currentNode -> Child" lines
+        try {
+            if (node.getParent() == null || node.getParent() == node) {
+                writer.write("digraph \"" + node.toNodeString() + "\" { " + System.lineSeparator());
+            }
 
-        List<DependencyNode> children = node.getChildren();
+            // Generate "currentNode -> Child" lines
 
-        for (DependencyNode child : children) {
-            writer.println("\t\"" + node.toNodeString() + "\" -> \"" + child.toNodeString() + "\" ; ");
+            List<DependencyNode> children = node.getChildren();
+            for (DependencyNode child : children) {
+                writer.write("\t\"" + node.toNodeString() + "\" -> \"" + child.toNodeString() + "\" ; "
+                        + System.lineSeparator());
+            }
+            writer.flush();
+            return true;
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to write to DOT output", e);
```

Comment:
[UncheckedIOException](https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html) but really we need to figure out how to not thrown an exception at all here, or perhaps  change the method signature. Possibly we need to put everything in a StringWriter first and then write the string later.